### PR TITLE
Pipe output in realtime

### PR DESCRIPTION
**Pull request by lhoste-bell**

### CODE REVIEW

**Comment by smarr** [comment]
Hm, this is problematic, because you always print standard out, unconditional of whether rebench is run in debug mode or with verbose flag. Do I see this right? I'd rather have a way to configure that by the user. Because I don't want gigabytes of log files on my CI system.

**Comment by lhoste-bell** [comment]
That's right, should I reuse the `-verbose` flag?

**Comment by smarr** [comment]
I think at the moment most stuff is somehow tied to the debug flag. The verbose flag is not actually used, except in one place as far as grep tells me.

So, I guess, verbose would be ok.

**Comment by lhoste-bell** [comment]
Bad idea. When a run fails, the output is print [twice](https://github.com/smarr/ReBench/blob/04fdf366b1e07454296bd3daaf42a46183f0020a/rebench/reporter.py#L139).
Can I remove the [`self._stdout == PIPE`](https://github.com/smarr/ReBench/pull/52/files#diff-b2eabfd3873eef4d1af23d8ec0f7ee9dR29) check? This makes the print depend on the single verbose flag in both cases.

**Comment by smarr** [comment]
I don't remember enough to be sure. Will need to look into it in more details. Can't promise to get to it today, sorry.

File: rebench/executor.py [hunk 1]
```diff
@@ -279,10 +280,11 @@ def _generate_data_point(self, cmdline, gauge_adapter, run_id,
                                                           stdout=subprocess.PIPE,
                                                           stderr=subprocess.STDOUT,
                                                           shell=True,
+                                                          verbose=self._verbose,
                                                           timeout=run_id.bench_cfg.suite.max_runtime)
         if return_code != 0 and not self._include_faulty:
             run_id.indicate_failed_execution()
-            run_id.report_run_failed(cmdline, return_code, output)
+            run_id.report_run_failed(cmdline, return_code, ("", output)[self._verbose == False])
```
Comment (smarr) [inline rebench/executor.py]: Oh man, really? that's intention preserving?

File: rebench/subprocess_with_timeout.py [hunk 2]
```diff
@@ -25,19 +26,30 @@ def run(self):
...
         self.pid = p.pid
 
-        self.stdout_result, self.stderr_result = p.communicate()
+        if self._verbose and self._stdout == PIPE:
+            self.stdout_result = ""
+            while True:
+                line = p.stdout.readline()
+                if not line:
+                    break
+                print line.rstrip()
```
Comment (smarr) [inline rebench/subprocess_with_timeout.py]: I don't think we should strip here

**Comment by smarr** [comment]
Ok, reworked it a little. Hope this works as expected.

**Pull request merged**